store/store.go

```diff
@@ -188,6 +211,13 @@ func generateMongoQuery(p *Params) bson.M {
 	}
 
 	andQuery := []bson.M{}
+	if p.UploadId != "" {
+		uploadIDQuery := []bson.M{
+			{"uploadId": p.UploadId},
+		}
+		andQuery = append(andQuery, bson.M{"$or": uploadIDQuery})
```

Comment:
The `$or` isn't adding any value here (since you only have one query item inside the `$or`). Suggest that you delete the above 3 lines, and change this line to:
```suggestion
		andQuery = append(andQuery, bson.M{"uploadId": p.UploadId})
```

Reply:
Right.  Makes sense.